src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountUpdateActionUtilsTest.java

```diff
@@ -0,0 +1,1473 @@
+        when(oldCartDiscount.getStackingMode()).thenReturn(null);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getStackingMode()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> changeStackingModeUpdateAction =
+                buildChangeStackingModeUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(changeStackingModeUpdateAction).contains(ChangeStackingMode.of(StackingMode.STACKING));
+    }
+
+    @Test
+    void buildSetValidFromUpdateAction_WithDifferentValues_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getValidFrom()).thenReturn(ZonedDateTime.parse("2019-04-30T22:00:00.000Z"));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        final ZonedDateTime now = ZonedDateTime.now();
+        when(newCartDiscountDraft.getValidFrom()).thenReturn(now);
+
+        final Optional<UpdateAction<CartDiscount>> setValidFromUpdateAction =
+                buildSetValidFromUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(setValidFromUpdateAction).contains(SetValidFrom.of(now));
+    }
+
+    @Test
+    void buildSetValidFromUpdateAction_WithSameValues_ShouldNotBuildUpdateAction() {
+        final ZonedDateTime validFrom = ZonedDateTime.parse("2019-04-30T22:00:00.000Z");
+
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getValidFrom()).thenReturn(validFrom);
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getValidFrom()).thenReturn(validFrom);
+
+        final Optional<UpdateAction<CartDiscount>> setValidFromUpdateAction =
+                buildSetValidFromUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(setValidFromUpdateAction).isNotPresent();
+    }
+
+    @Test
+    void buildSetValidFromUpdateAction_WithJustNewSetValidFromDateHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getValidFrom()).thenReturn(ZonedDateTime.parse("2019-04-30T22:00:00.000Z"));
+
+        final CartDiscountDraft newCartDiscountDraft = mock(CartDiscountDraft.class);
+        when(newCartDiscountDraft.getValidFrom()).thenReturn(null);
+
+        final Optional<UpdateAction<CartDiscount>> setValidFromUpdateAction =
+                buildSetValidFromUpdateAction(oldCartDiscount, newCartDiscountDraft);
+
+        assertThat(setValidFromUpdateAction).contains(SetValidFrom.of(null));
+    }
+
+    @Test
+    void buildSetValidFromUpdateAction_WithJustOldSetValidFromDateHasNullValue_ShouldBuildUpdateAction() {
+        final CartDiscount oldCartDiscount = mock(CartDiscount.class);
+        when(oldCartDiscount.getValidFrom()).thenRetu
```

Comment:
Please add a test for testing same values of different instances (which would be the more common case actually happening)